node/queue.go

```diff
@@ -146,7 +146,7 @@ func (pc *PodController) providerSyncLoop(ctx context.Context, q workqueue.RateL
 func (pc *PodController) runSyncFromProvider(ctx context.Context, q workqueue.RateLimitingInterface) {
 	if pn, ok := pc.provider.(PodNotifier); ok {
 		pn.NotifyPods(ctx, func(pod *corev1.Pod) {
-			enqueuePodStatusUpdate(ctx, q, pod)
+			pc.enqueuePodStatusUpdate(ctx, q, pod.DeepCopy())
```

Comment:
We end up calling `DeepCopy` in eneuquePodStatusUpdate as well.

Reply:
I moved to doing the `DeepCopy` before `enqueuePodStatusUpdate`